jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -93,25 +96,39 @@ def on_mouse_or_key_event(self, data):
             self.label_mouseover.world = ""
             self.label_mouseover.value = ""
 
-        if data['event'] == 'keydown' and data['key'] == 'b':
-
-            # Simple blinking of images - this will make it so that only one
-            # layer is visible at a time and cycles through the layers.
-
-            if len(self.state.layers) > 1:
-
-                # If only one layer is visible, pick the next one to be visible,
-                # otherwise start from the last visible one.
-
-                visible = [ilayer for ilayer, layer in
-                           enumerate(self.state.layers) if layer.visible]
-
-                if len(visible) > 0:
-                    next_layer = (visible[-1] + 1) % len(self.state.layers)
-                    self.state.layers[next_layer].visible = True
-                    for ilayer in visible:
-                        if ilayer != next_layer:
-                            self.state.layers[ilayer].visible = False
+        elif data['event'] == 'keydown' and data['key'] == 'b':
+            self.blink_once()
+
+    def blink_once(self):
+        # Simple blinking of images - this will make it so that only one
+        # layer is visible at a time and cycles through the layers.
+
+        # Exclude Subsets: They are global
+        valid = [ilayer for ilayer, layer in enumerate(self.state.layers)
+                 if isinstance(layer.layer, BaseData)]
+        n_layers = len(valid)
+
+        if n_layers == 1:
+            msg = SnackbarMessage(
+                'Nothing to blink', color='warning', sender=self)
+            self.session.hub.broadcast(msg)
+
+        elif n_layers > 1:
+            # If only one layer is visible, pick the next one to be visible,
+            # otherwise start from the last visible one.
+
+            visible = [ilayer for ilayer in valid if self.state.layers[ilayer].visible]
+            n_visible = len(visible)
+
+            if n_visible == 0:
+                msg = SnackbarMessage('No visible layer to blink',
```

Comment:
I actually don't know under what condition we will fall into this block but doesn't hurt?